python/basix/ufl.py

```diff
@@ -1398,38 +1368,124 @@ def _create_enriched_element(elements: _typing.List[_BasixElementBase],
+        gdim: Geometric dimension. If not set the geometric dimension is
+            set equal to the topological dimension of the cell.
+    """
+    return BasixElement(_basix.create_custom_element(
+        cell_type, list(value_shape), wcoeffs, x, M, interpolation_nderivs,
+        map_type, sobolev_space, discontinuous, highest_complete_degree, highest_degree), gdim=gdim)
+
+
+def mixed_element(elements: _typing.List[_ElementBase], gdim: _typing.Optional[int] = None) -> _ElementBase:
+    """Create a UFL compatible mixed element from a list of elements.
+
+    Args:
+        elements: The list of elements
+        gdim: Geometric dimension. If not set the geometric dimension is
+            set equal to the topological dimension of the cell.
+    """
+    return MixedElement(elements, gdim=gdim)
+
+
+def blocked_element(sub_element: _ElementBase, rank: _typing.Optional[int] = None,
+                    shape: _typing.Optional[_typing.Tuple[int, ...]] = None,
+                    symmetry: _typing.Optional[bool] = None, gdim: _typing.Optional[int] = None) -> _ElementBase:
+    """Create a UFL compatible blocked element.
+
+    Args:
+        sub_element: The element used for each block.
+        rank: Rank of the value shape of the element. For scalar-valued
+            families, setting `rank=1` creates a vector element where
+            each component of the vector is represented by the element.
+            Similarly, `rank>2` creates a tensor element. If `shape` is
+            provided, then `rank` is not required .
+        shape: Value shape of the element. For scalar-valued families,
+            this can be used to create vector and tensor elements. If
+            `rank` is set but `shape` is not, then `shape` will be set
+            to `()` for rank 0, `(tdim, )` for rank 1, and `(tdim, ...,
+            tdim)` for rank 2 or higher.
+        symmetry: Set to `True` if the tensor is symmetric. Valid for
+            rank 2 elements only.
+        gdim: Geometric dimension. If not set the geometric dimension is
+            set equal to the topological dimension of the cell.
+    """
+    if len(sub_element.value_shape()) != 0:
+        raise ValueError("Cannot create a blocked element containing a non-scalar element.")
+    if rank is None and shape is None:
+        raise ValueError("At least one of rank and shape must be set.")
+
+    if shape is None:
+        if rank is None:
+            shape = tuple(sub_element.value_shape())
+        else:
+            tdim = len(_basix.topology(sub_element.cell_type)) - 1
+            shape = tuple(tdim for _ in range(rank))
+    if rank is None:
+        rank = len(shape)
+    if rank != len(shape):
+        raise ValueError("Incompatible rank and shape.")
+
+    return BlockedElement(sub_element, shape=shape, symmetry=symmetry, gdim=gdim)
+
+
+def convert_ufl_element(ufl_element: _FiniteElementBase) -> _ElementBase:
```

Comment:
short-term convenience. It's needed so that we can handle elements created UFL, so will exist until the UFL work gets far enough to make it no longer needer.

(It gives a deprecation warning to encourage users to stop using it)